infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -145,11 +158,23 @@ private void handleLrEntryRequest(@Nonnull RequestMsg request,
 
             LogReplicationSinkManager sinkManager = sessionToSinkManagerMap.get(session);
 
-            // If no sink Manager is found, drop the message and log an error
+            // We create a sinkManager for sessions that are discovered while bootstrapping LR. But as topology changes,
+            // we may discover new sessions. At the same time, its possible that the remote Source cluster finds a new
+            // session before the local cluster and sends a request to the local cluster.
+            // Since the two events are async, we wait to receive a new session in the incoming request.
+            // If the incoming session is not known to sessionManager drop the message (until session is discovered by
+            // local cluster), otherwise create a corresponding sinkManager.
+            // TODO[V2] : We still have a case where the cluster does not ever discover a session on its own.
+            //  To resolve this, we need to have a long living RPC from the connectionInitiator cluster which will query
+            //  for sessions from the other cluster
             if (sinkManager == null) {
-                log.error("Sink Manager not found for session {}, total={}, sessions={}", session,
-                        sessionToSinkManagerMap.size(), sessionToSinkManagerMap.keySet());
-                return;
+                if(!sessionManager.getSessions().contains(session)) {
```

Comment:
personally I think we can do
```
if (!sessionToSinkManagerMap.containsKey(session)) {
    xxxx
    createSinkManager(session)
}
sinkManager = sessionToSinkManagerMap.get(session);
```
since `createSinkManager` sounds more like a void method to me. But just personal favor, no strong preference